### PR TITLE
feat: add Dockerfile

### PR DESCRIPTION
**Pull request by mudler**
See: https://github.com/SUSE/eirinix-helm-release/pull/3

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ARG email=ci-ci-bot@suse.de
+ADD . /eirini-ssh
+WORKDIR /eirini-ssh
+RUN git config --global user.name ${user}
+RUN git config --global user.email ${email}
+RUN GO111MODULE=on go mod vendor
+RUN bin/build-extension
+RUN bin/build-ssh-proxy
+
+FROM bitnami/kubectl as kubectl
```
Comment (mook-as) [inline Dockerfile:13]: Would it be possible to grab `kubectl` directly from [upstream] directly? I'm a bit queasy with how difficult it is to track down the source of the bitnami images.

We have a [kubernetes-client] package in OBS, but that's not built with `CGO_ENABLED=0` so it is (probably) dynamically linked and requires using the appropriate base image.

[upstream]: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.18.md#client-binaries
[kubernetes-client]: https://software.opensuse.org/package/kubernetes-client

**Comment by mudler** [comment]
It doesn't need `kubectl` as far I can remember, but we bundled it ( @jimmykarily  correct me If I recall wrong ) to help while troubleshooting while developing . For consistency I would keep it like we do in the loggregator-bridge https://github.com/SUSE/eirini-loggregator-bridge/blob/master/Dockerfile#L14 ,  I'll try to find a way to make it optional

**Comment by jimmykarily** [comment]
kubectl shouldn't be needed by the binaries. I also think we added that for debugging.

**Comment by mudler** [comment]
I've made kubectl optional, and fetching it now directly upstream, added also sha512 checks:

```
docker build --build-arg DEBUG_TOOLS=true --build-arg BASE_IMAGE=scratch  -t eirini-ssh .
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
+ARG BASE_IMAGE=opensuse/leap
+
+FROM golang:1.12 as build
+ARG user="SUSE CFCIBot"
+ARG email=ci-ci-bot@suse.de
```
Comment (mook-as) [inline Dockerfile:5]: For consistency, upper case?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ARG KUBECTL_CHECKSUM=ed36f49e19d8e0a98add7f10f981feda8e59d32a8cb41a3ac6abdfb2491b3b5b3b6e0b00087525aa8473ed07c0e8a171ad43f311ab041dcc40f72b36fa78af95
+ADD . /eirini-ssh
+WORKDIR /eirini-ssh
+RUN git config --global user.name ${user}
+RUN git config --global user.email ${email}
+RUN if [ "$DEBUG_TOOLS" = "true" ] ; then \
+    wget -O kubectl.tar.gz https://dl.k8s.io/$KUBECTL_VERSION/kubernetes-client-$KUBECTL_ARCH.tar.gz && \
+    echo "$KUBECTL_CHECKSUM kubectl.tar.gz" | sha512sum --check --status && \
+    tar xvf kubectl.tar.gz -C / && \
+    cp -rf /kubernetes/client/bin/kubectl /eirini-ssh/binaries/; fi
```
Comment (mook-as) [inline Dockerfile:18]:
```suggestion
    cp -f /kubernetes/client/bin/kubectl /eirini-ssh/binaries/; fi
```

**Pull request merged**